### PR TITLE
Add integration specs fix n+1

### PR DESCRIPTION
**Pull request by HeDevedUp**
For this milestone, we:

Added integration specs to:
Users list (main page).
Each user profile.
User's list of posts.
Each user's individual post.
Fixed N+1 problems on posts #show & posts #index.
Added bullet & database_cleaner gems.

### CODE REVIEW

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!



The following test are have not been added for this project to be approved 

#### Has featured tests for `User` **index page**:

* Tests that assert you can see the username of all other users.
* Tests that assert you can see the profile picture for each user.
* Tests that assert you can see the number of posts each user has written.
* Tests that redirect to the user's show page when you click on a user.

#### Has featured tests for `User` show page:

* Tests that assert you can see the user's profile picture.
* Tests that asserts you can see the user's username.
* Tests that assert you can see the number of posts the user has written.
* Tests that asserts you can see the user's bio.
* Tests that asserts you can see the user's first 3 posts.
* Tests that assert you can see a button that lets me view all of a user's posts.
* Tests that redirect to the post's show page when you click on a user's post.
* Tests that redirect to the user's post's index page when you click on a **see all posts**.

#### Has featured tests for `User` **post index page**:

* Tests that assert you can see the user's profile picture.
* Tests that asserts you can see the user's username.
* Tests that assert you can see the number of posts the user has written.
* Tests that asserts you can see a post's title.
* Tests that assert you can see some of the post's body.
* Tests that asserts you can see the first comments on a post.
* Tests that assert you can see how many comments a post has.
* Tests that assert you can see how many likes a post has.
* Tests that assert you can see a section for pagination if there are more posts than fit on the view.
* Tests that redirect to the post's show page when you click on a post.

#### Has featured tests for `User` **post show page**:

* Tests that assert you can see the post's title.
* Tests that assert you can see who wrote the post.
* Tests that assert you can see how many comments it has.
* Tests that assert you can see how many likes it has.
* Tests that assert you can see the post body.
* Tests that assert you can see the username of each comment.
* Tests that assert you can see the comment each comment left.



## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: spec/views/users/index.html.erb_spec.rb [hunk 1]
```diff
@@ -1,5 +1,33 @@
...
+    user2 = User.create(
+      name: 'Second User',
+      photo: 'https://www.pexels.com/photo/silhouette-of-a-person-on-a-swing-3293148/',
+      bio: "I'm not a student at Microverse",
+      posts_counter: 0
+    )
+    @users = [user1, user2]
+    visit '/users'
+  end
+
```
Comment (tufoinnkuo10) [inline spec/views/users/index.html.erb_spec.rb:38]: - [ ] kindly add Tests that asserts you can see the username of all other users.

File: spec/views/posts/index.html.erb_spec.rb [hunk 2]
```diff
@@ -1,5 +1,58 @@
...
+    )
+    visit "/users/#{@user.id}"
+  end
+
+  it 'displays user info' do
+    expect(page).to have_content(@user.name)
+    expect(page).to have_css("img[src*='#{@user.photo}']")
+    expect(page).to have_content(@user.posts_counter.to_s)
+  end
+
```
Comment (tufoinnkuo10) [inline spec/views/posts/index.html.erb_spec.rb:33]: - [ ] kindly add Tests that asserts you can see the number of posts the user has written.

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @HeDevedUp 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally

Required Changes ♻️
Check the comments under the review
Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by andersonlebon — APPROVED** [review]
## Hi @HeDevedUp 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Optional suggestions
- [ ] Kindly Check the comments under the review._
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**